apis/python/src/tiledbsoma/_general_utilities.py

```diff
@@ -52,6 +53,11 @@ def get_implementation_version() -> str:
             return "unknown"
 
 
+def get_release_version() -> Tuple[int, int]:
+    version = get_implementation_version().split(".")
+    return (int(version[0]), int(version[1]))
```

Comment:
This is going to fail if the `get_implementation_version` function fails. Since this is now called only once, I would inline it into the `_tiledb_ctx_deprecation` function and if it does fail, just skip the assertion and issue the warning, so that if somebody’s setup metadata gets messed up it doesn’t break SOMA entirely.

Reply:
I actually just created a [new PR](https://github.com/single-cell-data/TileDB-SOMA/pull/2701) on top of this one that uses `get_implementation_version` in another file. Should I inline it for there as well or do better error handling for this

Reply:
I would just add better error handling here—it might be as simple as returning `(0, 0)`, or perhaps this could be written as a function like `assert_version_before(major: int, minor: int)`, so that the assert code is in one place